src/main/java/it/cnr/isti/hpc/wikipedia/article/Link.java

```diff
@@ -50,19 +53,18 @@ public void setId(String id) {
 	}
 	
 	
-	public String getDescription() {
-		return this.description;
+	public String getAnchor() {
+		return this.anchor;
 	}
 	
 	
-	public void setDescription(String description) {
+	public void setAnchor(String anchor) {
         // Some links do not have any anchor
         // For those cases the anchor is the same wikipedia Id
-		if (description.isEmpty()){
-            this.description = this.id.replace("_", " ");
-        }else{
-            this.description = description;
-        }
+        if (anchor == null || anchor.isEmpty())
+           this.anchor = this.id.replace("_", " ");
```

Comment:
No, this is just handling annotations without anchors like [[Barack_Obama]]
For some reason whatever component generates them, was sending also `nulls` so when you built the jsonpedia it would output some exceptions.
Thus the extra check was added